chat.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ably/ably-go/ably"
+	"github.com/ably/ably-go/ably/proto"
+)
+
+func main() {
+	fmt.Println("Type your clientID")
+	reader := bufio.NewReader(os.Stdin)
+	clientId, _ := reader.ReadString('\n')
+	clientId = strings.Replace(clientId, "\n", "", -1)
+
+	opts := &ably.ClientOptions{
+		AuthOptions: ably.AuthOptions{
+			// If you have an Ably account, you can find
+			// your API key at https://www.ably.io/accounts/any/apps/any/app_keys
+			Key: "INSERT_YOUR_API_KEY_HERE",
```

Comment:
We shouldn't encourage people to hard code the api key. It should be an env var or a cli param.We use github.com/urfave/cli/v2 for our tools.

Reply:
I that worth it for a demo though? Could be using a const string at the top of the file maybe?

Reply:
If we tell customers not to do it then we shouldn't do it. Doing it as an env var is really simple and a cli is pretty straightforward.
```
Key: os.GetEnv("ABLY_KEY")
```

Reply:
Ah yes if we say that then it makes sense.